cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationRollingPolicyBase.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.plugins;
+
+import ch.qos.logback.core.FileAppender;
+import ch.qos.logback.core.rolling.helper.CompressionMode;
+import ch.qos.logback.core.rolling.helper.FileNamePattern;
+import ch.qos.logback.core.spi.ContextAwareBase;
+import org.apache.twill.filesystem.Location;
+
+import java.io.Closeable;
+
+
+/**
+ * Location rolling policy base which provides
+ */
+public abstract class LocationRollingPolicyBase extends ContextAwareBase implements LocationRollingPolicy {
+
+  protected CompressionMode compressionMode = CompressionMode.NONE;
```

Comment:
Generally, better to have fields as `private` and expose getter for sub-classes to gain access to the field rather than using the field directly, especially these fields are non-final.